scripts/utils.js

```diff
@@ -28,3 +30,48 @@ export function validJSON(string) {
 export function cleanRecord(record) {
   return omit(record, ["id", "schema", "last_modified"]);
 }
+
+export function recordField(displayField, record) {
+  if (record.hasOwnProperty(displayField)) {
+    const field = record[displayField];
+    if (typeof field === "string") {
+      return field;
+    } else if (typeof field === "object") {
+      return JSON.stringify(field);
+    } else {
+      return String(field);
+    }
+  } else if (displayField === "__json") {
+    return <code>{JSON.stringify(cleanRecord(record))}</code>;
+  } else if (displayField.indexOf(".") !== -1) {
+    const fields = displayField.split(".");
+
+    if (record.hasOwnProperty(fields[0])) {
+      return recordField(fields.splice(1).join("."), record[fields[0]]);
+    } else {
+      let biggestCandidate = [];
+      let nextCandidate = [];
+      let candidates = Object.keys(record).filter((key) => {
+        return key.indexOf(fields[0]) === 0;
+      });
+
+      for (let key in candidates) {
+        let nextCandidate = [];
+        for (let part of fields) {
+          var candidate = nextCandidate.concat([part]).join(".");
+          if (candidates[key].indexOf(nextCandidate.concat([part]).join(".")) !== -1) {
+            nextCandidate.push(part);
+          }
+        }
+        if (nextCandidate.length > biggestCandidate.length) {
+          biggestCandidate = nextCandidate;
+        }
+      }
+
+      const key = biggestCandidate.join(".");
+      return recordField(fields.splice(biggestCandidate.length).join("."),
+                         record[key]);
```

Comment:
I'm having a tough time figuring out the actual intent of this code, so we might want to add a few comments there.